src/python_inspector/utils_pypi.py

```diff
@@ -1631,7 +1645,18 @@ def get_remote_file_content(
         echo_func = print
     if verbose:
         echo_func(f"DOWNLOADING: {url}")
-    with requests.get(url, allow_redirects=True, stream=True, headers=headers) as response:
+    if credentials:
+        stream = requests.get(
```

Comment:
I think you can set auth in the if statements, and then call `requests.get` afterwards

```suggestion
        auth = (credentials.get("login"), credentials.get("password"))
    else:
        auth = None
    stream = requests.get(
        url,
        allow_redirects=True,
        stream=True,
        headers=headers,
        auth=(credentials.get("login"), credentials.get("password")),
    )
```